packages/svelte/src/internal/client/dom/blocks/boundary.js

```diff
@@ -188,6 +200,13 @@ export class Boundary {
 		return !!this.#props.pending;
 	}
 
+	get_batch() {
+		if (current_batch) {
+			this.#batch = current_batch;
+		}
+		return /** @type {Batch} */ (this.#batch);
```

Comment:
this is the only thing in this PR that gives me a bit pause, basically that we have no definite way to know which batch is the latest one for which boundary, this "only" feels like a good enough approximation.